srml/support/src/dispatch.rs

```diff
@@ -188,6 +181,31 @@ macro_rules! decl_module {
 			$($rest)*
 		);
 	};
+	(@normalize
+		$(#[$attr:meta])*
+		pub struct $mod_type:ident<$trait_instance:ident: $trait_name:ident>
+		for enum $call_type:ident where origin: $origin_type:ty, system = $system:ident
+		{ $( $deposit_event:tt )* }
+		{ $( $on_finalise:tt )* }
+		[ $($t:tt)* ]
+		$(#[doc = $doc_attr:tt])*
+		$fn_vis:vis fn $fn_name:ident($origin:ident : T::Origin $(, $param_name:ident : $param:ty)* ) -> $result:ty { $( $impl:tt )* }
```

Comment:
Do we really want to match on `T::Origin` here? 
We already give the user the functionality to specify the the origin type. So, the type name could be something different than `T::Origin` and this match would not work here.

Reply:
true but this is a special type, so we would have no way of identifying it. this is to catch the case where the user accidentally does `fn foo(origin: T::Origin, bar: Bar)` instead of `fn foo(origin, bar: Bar)` (which already happened once and caused significant developer downtime).

i'm also happy with throwing a build error.

Reply:
i'll do both, then we catch `origin: Whatever` and `whatever: T::Origin`